### PR TITLE
Address spurious Microsoft compiler warnings

### PR DESCRIPTION
**Pull request by dan4thewin**
Different compilers seem to have different capacities to recognize use of an uninitialized variable.  The GNU toolchain gives no warnings, but the Microsoft compiler does.  Upon inspection, the variables are only ever used after being set, but it is simpler to provide throwaway defaults than to mute the warning.

### CODE REVIEW

File: source/core_json.c [hunk 1]
```diff
@@ -1248,7 +1248,7 @@ static JSONStatus_t search( char * buf,
                             size_t * outValueLength )
 {
     JSONStatus_t ret = JSONPartial;
-    size_t i = 0, key, keyLength, value, valueLength;
+    size_t i = 0, key, keyLength, value = 0, valueLength = 0;
```
Comment (jainamt) [inline source/core_json.c:1251]: what abt ```key``` and ```keyLength```?

Comment (dan4thewin) [reply source/core_json.c:1251]: There are no warnings for key and keyLength.  I think it is better to enlist the compiler to tell me when I use an uninitialized variable, than to initialize everything and rob myself of that check.  The only problem is when the compiler gets it wrong, as the Microsoft compiler does here.

Comment (sarenameas) [reply source/core_json.c:1251]: Do you think that it's good practice to just initialize everything anyways as part of our style?

Comment (dan4thewin) [reply source/core_json.c:1251]: No, not everything.  I think the practice can do more harm than good in some cases.

**Review by sarenameas — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**